src/cc65/symentry.h

```diff
@@ -245,10 +248,10 @@ INLINE int SymIsRegVar (const SymEntry* Sym)
 /* Return true if the given entry is a register variable */
 /* ### HACK! Fix the ugly type flags! */
 {
-    return ((Sym->Flags & (SC_REGISTER|SC_TYPE)) == SC_REGISTER);
+    return ((Sym->Flags & (SC_REGISTER|SC_TYPEMASK)) == SC_REGISTER);
 }
 #else
-#  define SymIsRegVar(Sym)      (((Sym)->Flags & (SC_REGISTER|SC_TYPE)) == SC_REGISTER)
+#  define SymIsRegVar(Sym)  (((Sym)->Flags & (SC_REGISTER|SC_ESUTYPEMASK)) == SC_REGISTER)
```

Comment:
The inline function and the macro must do the same test!

Reply:
The macro version should use `SC_TYPEMASK` instead of `SC_ESUTYPEMASK`. Fixed.